### PR TITLE
OrganizationUnit should have filter parameter

### PR DESCRIPTION
**Pull request by jonny-xhl**
- I think OrganizationUnit should have filter parameter.
- Add LongCount

### CODE REVIEW

File: modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreOrganizationUnitRepository.cs [hunk 1]
```diff
@@ -162,7 +168,18 @@ public override IQueryable<OrganizationUnit> WithDetails()
             DbContext.Set<IdentityUserOrganizationUnit>().RemoveRange(ouMembersQuery);
         }
 
-        protected virtual IQueryable<IdentityUser> CreateGetMembersFilteredQuery(OrganizationUnit organizationUnit, string filter = null)
+        public virtual async Task<long> GetLongCountAsync(Guid? parentId, string filter = null,
```
Comment (gterdem) [inline modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreOrganizationUnitRepository.cs:255]: Why having parentId parameter when it's not been used?

File: modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/IOrganizationUnitRepository.cs [hunk 2]
```diff
@@ -80,5 +81,10 @@ public interface IOrganizationUnitRepository : IBasicRepository<OrganizationUnit
             OrganizationUnit organizationUnit,
             CancellationToken cancellationToken = default
         );
+
+        Task<long> GetLongCountAsync(
+            Guid? parentId,
```
Comment (gterdem) [inline modules/identity/src/Volo.Abp.Identity.Domain/Volo/Abp/Identity/IOrganizationUnitRepository.cs:119]: Is `parentId` parameter necessary?

File: modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoOrganizationUnitRepository.cs [hunk 3]
```diff
@@ -155,7 +161,18 @@ public virtual async Task RemoveAllMembersAsync(OrganizationUnit organizationUni
             }
         }
 
-        protected virtual IMongoQueryable<IdentityUser> CreateGetMembersFilteredQuery(OrganizationUnit organizationUnit, string filter = null)
+        public virtual async Task<long> GetLongCountAsync(Guid? parentId, string filter = null,
```
Comment (gterdem) [inline modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoOrganizationUnitRepository.cs:236]: Same unused `parentId` parameter.

**Review by gterdem — CHANGES_REQUESTED** [review]
Thank you for your contribution. 
`GetLongCountAsync` method has some unused parameters. Would you mind to check it?

File: modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoOrganizationUnitRepository.cs (hunk 3 above)
Comment (jonny-xhl) [reply modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoOrganizationUnitRepository.cs:236]: I have already solved.

1. I think it is necessary for GetListAsync to filter through parentId when the OrgnizationUnit is more complex.

**Example:**
- The organization structure tree is lazily loaded.
- When there are many suborganizational units.

Comment (gterdem) [reply modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoOrganizationUnitRepository.cs:236]: I can't see updated code on View Change set. Can you update your dev branch?

Comment (jonny-xhl) [reply modules/identity/src/Volo.Abp.Identity.MongoDB/Volo/Abp/Identity/MongoDB/MongoOrganizationUnitRepository.cs:236]: lasted commit id 79f1369(resolved conflicts).

**Review by gterdem — APPROVED** [review]
Thank you for your contribution.

**Comment by maliming** [comment]
hi

When the `parentId `parameter of the `GetListAsync `and `GetLongCountAsync `methods is **null**, we are actually querying the root node. Is this expected?

Is this a breaking change?

**Comment by ismcagdas** [comment]
@maliming I think this is a breaking change. Maybe we shouldn't add parentId to filter if parentId is null. But in that case, there is no way to just query root items.

**Comment by gterdem** [comment]
@maliming thanks. I forgot to predict the null use case.

**Pull request merged**